src/Cli/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddJsonPropertyPostActionProcessor.cs

```diff
@@ -12,6 +13,8 @@ namespace Microsoft.TemplateEngine.Cli.PostActionProcessors
 {
     internal class AddJsonPropertyPostActionProcessor : PostActionProcessorBase
     {
+        private const string AllowFileCreationArgument = "allowFileCreation";
+        private const string AllowPathCreationArgument = "allowPathCreation";
```

Comment:
Split into 2 properties to allow users to be able to add the missing property path without forcing requirement to allow file creation.

There is some potential discussion here because it's not really making sense to allow file creation and not allow path creation so we could maybe have some kind of enum instead (e.g. `CreationMode` with `None` being the default, `Path` and `File` or `FileAndPath`). We could also keep the boolean and say the `allowPathCreation` is assumed to be `true` when `allowFileCreation` is set to `true`.